src/main/java/com/hierynomus/smbj/utils/SmbFiles.java

```diff
@@ -64,4 +74,31 @@ public static int copy(File source, DiskShare share, String destPath, boolean ov
         return r;
     }
 
+    /**
+     * Create a set of nested sub-directories in the given path, for example, 2345 \ 3456 \ 4453 \ 123123.txt
+     */
+    public void mkdirs(DiskShare diskShare, String path) throws SMBApiException {
```

Comment:
You can simplify this greatly by inverting the logic. Instead of starting from the beginning, find the longest path segment that exists and only create the tail end.

For deeply nested paths for which a big part already exists, this will save a lot of remote calls.

Reply:
I'm sorry, but I can’t determine how to implement this. Can you help me to fix this issue ?

Reply:
why not use static method for mkdirs?